provider/gateway/rest/client.go

```diff
@@ -407,6 +518,20 @@ func (c *client) ServiceLogs(ctx context.Context,
 		OnClose: onclose,
 	}
 
+	if err = conn.SetReadDeadline(time.Now().Add(pingWait)); err != nil {
+		onclose <- parseCloseMessage(err.Error())
+		return nil, err
+	}
+
+	conn.SetPingHandler(func(string) error {
```

Comment:
This looks identical to another ping handler, can this be made a struct with the needed variables & a receiver function?